cupy/lib/_routines_poly.py

```diff
@@ -71,3 +71,43 @@ def polysub(a1, a2):
         out = out.astype(cupy.result_type(a1, a2), copy=False)
         out -= 2 * out - a1
     return out
+
+
+def roots(p):
+    """Computes the roots of a polynomial with given coefficients.
+
+    Args:
+        p (cupy.ndarray): polynomial coefficients.
+
+    Returns:
+        cupy.ndarray: polynomial roots.
+
+    .. warning::
+
+        This function doesn't support currently polynomial coefficients
+        whose companion matrices are general 2d square arrays. Only those
+        with complex Hermitian or real symmetric 2d arrays are allowed.
+
+    .. seealso:: :func:`numpy.roots`
+
+    """
+    if p.ndim == 0:
+        raise TypeError('0-dimensional input is not allowed')
+    if p.size < 2 or (p.ndim == 1 and not cupy.any(p)):
```

Comment:
When `p.ndim == 1 and not cupy.any(p)`, the condition in `L99` seems evaluated to `True`.
```suggestion
    if p.size < 2:
```